packages/synapse-react-client/src/components/FeaturedDataTabs/FacetPlotsCard.tsx

```diff
@@ -49,19 +59,59 @@ const layout: Partial<Plotly.Layout> = {
   },
 }
 
+function LoadingCard(props: { numPlots: number }) {
+  const { numPlots } = props
+  return (
+    <Paper className="FacetPlotsCard">
+      <FacetPlotsCardTitleContainer
+        className={FACET_PLOTS_CARD_TITLE_CONTAINER_CLASSNAME}
+      >
+        <Skeleton
+          className="FacetPlotsCard__title"
+          width={'60%'}
+          height={'24px'}
+        />
+
+        <SkeletonParagraph
+          className="FacetPlotsCard__description"
+          numRows={5}
+        />
+
+        <Skeleton width={'40%'} className={'FacetPlotsCard__link'}>
+          <Button variant={'contained'}>Explore</Button>
+        </Skeleton>
+      </FacetPlotsCardTitleContainer>
+      {times(numPlots).map(index => (
+        <FacetPlotsCardPlotContainer
+          key={index}
+          className={FACET_PLOTS_CARD_PLOT_CONTAINER_CLASSNAME}
+          sx={{
+            py: 3,
+            gridRow: `plot${index}`,
+          }}
+        >
+          <Skeleton width={'100%'} height={'300px'} />
+          <SkeletonTable numRows={4} numCols={2} />
+        </FacetPlotsCardPlotContainer>
+      ))}
+    </Paper>
+  )
+}
 const FacetPlotsCard: React.FunctionComponent<FacetPlotsCardProps> = ({
   title,
   description,
   facetsToPlot,
   detailsPagePath,
 }: FacetPlotsCardProps): JSX.Element => {
   const { accessToken } = useSynapseContext()
-  const { data, isLoadingNewBundle } = useQueryContext()
-  const { getColumnDisplayName, rgbIndex } = useQueryVisualizationContext()
+  const { data, isLoadingNewBundle } = useQueryContext<
+    'columnModels' | 'facets',
+    true
```

Comment:
I added some generics to make using `useQueryContext` a bit more friendly. Since the `partMask` is hard-coded, we can safely set this generic to basically say `columnModels` and `facets` will be defined. The boolean generic says to remove the other optional parts from the type of the returned object (e.g. the result rows).

Reply:
So it's still possible that the type is not aligned with the partMask.  In general, I think these generics will be helpful for compile-time confidence in the response.  However, if these do not match at runtime, a developer unfamiliar with this requirement will face an interesting mystery!